src/clients/HubPoolClient.ts

```diff
@@ -452,21 +454,41 @@ export class HubPoolClient {
         );
       }
     ) as ExecutedRootBundle;
-    if (mostRecentExecutedRootBundleEvent) {
-      // Arguably we don't need to even check these array lengths since we should assume that any proposed root bundle
-      // meets this condition.
-      if (
-        mostRecentExecutedRootBundleEvent.l1Tokens.length !== mostRecentExecutedRootBundleEvent.runningBalances.length
-      ) {
-        throw new Error("runningBalances and L1 token of ExecutedRootBundle event are not same length");
+
+    if (executedRootBundle) {
+      // runningBalances length must be a 1x or 2x multiple of l1Tokens length.
+      const { l1Tokens } = executedRootBundle;
+      let { runningBalances } = executedRootBundle;
+      let incentiveBalances: BigNumber[] = [];
+      if (runningBalances.length === l1Tokens.length) {
+        // Pre-UBA model (no incentives exist).
+        incentiveBalances = Array.from(runningBalances, () => toBN(0));
```

Comment:
Should we use `.map()` instead? I think it's a more common usage.
```suggestion
        incentiveBalances = runningBalances.map(() => toBN(0));
```

Reply:
Yeah this is fine. I was using map initially but I got complaints about an unused variable (i.e. the variable available in each iteration). Not defining anything is also a good way of avoiding that!